File: JavaScript_Courses/JavaScript_Basics/Preparation/04_For_Loop_Exercise/03_Histogram.js
function histogram(input) {
    //     < 200	
    // 200 … 399	
    // 400 … 599	
    // 600 … 799	
    // ≥ 800	
    let p1 = 0;
    let p2 = 0;
    let p3 = 0;
    let p4 = 0;
    let p5 = 0;

    let numbersCount = Number(input[0]);

    for (let i = 1; i <= numbersCount; i++) {
        let currentNumber = Number(input[i]);
        if (currentNumber < 200) {
            p1++;
        } else if (currentNumber < 400) {
            p2++;
        } else if (currentNumber < 600) {
            p3++;
        } else if (currentNumber < 800) {
            p4++;
        } else if (currentNumber >= 800) {
            p5++;
        }
    }

    console.log((p1 / numbersCount * 100).toFixed(2) + "%");
    console.log((p2 / numbersCount * 100).toFixed(2) + "%");
    console.log((p3 / numbersCount * 100).toFixed(2) + "%");
    console.log((p4 / numbersCount * 100).toFixed(2) + "%");
    console.log((p5 / numbersCount * 100).toFixed(2) + "%");
}

// histogram(["3", "1", "2", "999"]);

File: Python_Courses/Python_Basics/Python_PB_2025_03/04_For_Loop_Exercise/03_Histogram.py
p1_count = 0
p2_count = 0
p3_count = 0
p4_count = 0
p5_count = 0

numbers_count = int(input())

for _ in range(numbers_count):
    current_number = int(input())

    if current_number < 200:
        p1_count += 1
    elif current_number < 400:  # [200; 400) == [200; 399]
        p2_count += 1
    elif current_number < 600:
        p3_count += 1
    elif current_number < 800:
        p4_count += 1
    else:  # elif current_number >= 800:
        p5_count += 1

p1_percent = p1_count / numbers_count * 100
p2_percent = p2_count / numbers_count * 100
p3_percent = p3_count / numbers_count * 100
p4_percent = p4_count / numbers_count * 100
p5_percent = p5_count / numbers_count * 100

print(f'{p1_percent:.2f}%')
print(f'{p2_percent:.2f}%')
print(f'{p3_percent:.2f}%')
print(f'{p4_percent:.2f}%')
print(f'{p5_percent:.2f}%')
